services/early_settings/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+use core::sync::atomic::{AtomicU32, Ordering};
+use num_traits::*;
+use xous::{send_message, Message};
+
+pub const SERVER_NAME_ES: &str = "_EARLY_SETTINGS";
+
+#[derive(Debug, num_derive::FromPrimitive, num_derive::ToPrimitive)]
+pub enum Opcode {
+    /// Sets the keymap in early settings.
+    SetKeymap,
+
+    /// Retrieves keymap from early settings.
+    GetKeymap,
+
+    /// Retrieves the status of the early sleep flag.
+    EarlySleep,
+
+    /// Sets early sleep flag in early settings.
+    SetEarlySleep,
+}
+
+#[derive(Debug)]
+pub struct EarlySettings {
+    conn: xous::CID,
+}
+
+impl EarlySettings {
```

Comment:
I think it's worth re-iterating the warning about personal information and edge cases in the lib.rs file attached to this struct, because Rust autodoc will pick it up and ship it with the lib.rs?

If you're browsing source on GH, the README.md is what you see, but if you happen to be using autodocs to read things, the comment attached to this structure is what you'd see. We don't use autodoc right now but I could see it being a thing (and also probably people are trained to look here anyways for important messages about APIs).

Reply:
I wonder if I could include the README content as a autodoc 🤔 

In any case it’s a good idea!

Reply:
Yes, rustdoc will allow you to include files: https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html#the-doc-attribute

You can also use cfg variables to not document specific structs: https://doc.rust-lang.org/rustdoc/advanced-features.html#cfgdoc-documenting-platform-specific-or-feature-specific-information